edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -163,15 +163,15 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                     case Constants.IotedgedMode:
                         string managementUri = configuration.GetValue<string>(Constants.EdgeletManagementUriVariableName);
                         string workloadUri = configuration.GetValue<string>(Constants.EdgeletWorkloadUriVariableName);
-                        bool disableDeviceAnalyticsTelemetry = configuration.GetValue<bool>("DisableDeviceAnalyticsTelemetry", false);
+                        bool disableDeviceAnalyticsMetadata = configuration.GetValue<bool?>("DisableDeviceAnalyticsMetadata") ?? configuration.GetValue<bool>("DisableDeviceAnalyticsTelemetry", false);
```

Comment:
Did you test if this logic works as expected?

Reply:
Yes, I have a local project that I used to confirm this behavior.